core/src/commonMain/kotlin/com/xebia/functional/xef/llm/BaseChat.kt

```diff
@@ -0,0 +1,25 @@
+package com.xebia.functional.xef.llm
+
+import com.xebia.functional.xef.llm.models.MaxIoContextLength
+import com.xebia.functional.xef.llm.models.chat.Message
+
+interface BaseChat : LLM {
+
+  val contextLength: MaxIoContextLength
+
+  @Deprecated(
+    "will be removed from LLM in favor of abstracting former ModelType, use contextLength instead"
+  )
+  val maxContextLength
+    get() =
+      (contextLength as? MaxIoContextLength.Combined)?.total
+        ?: error(
+          "accessing maxContextLength requires model's context length to be of type MaxIoContextLength.Combined"
+        )
```

Comment:
As a side note, again - this is supposed to be an intermediary solution. Usages of this field that use an OAI model will still work as before. If this field is called on an instance of a (Google) model that doesn't use the combined context length an exception imminent.
I found this to be the best way to handle it, in favor to not change too much code in one PR.